### PR TITLE
ci: Add milestone workflow

### PR DESCRIPTION
**Pull request by paulgessinger**
This is a GitHub Actions based alternative to the milestone bot that we run. The actions setup will

- When a PR is opened or reopened, and has no milestone associated, it will assign the `next` milestone
- When a PR is milestoned or demilestoned, it will add a status to the PR like the current bot does it.

This should allow us to require a milestone to be set before merging, without having to host the bot anymore.

### CODE REVIEW

File: .github/workflows/milestone.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    if: ${{ github.event.issue.pull_request || github.event.pull_request }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # - run: |
+      #     echo "${{ toJSON(github.event) }}"
+
```
Comment (AJPfleger) [inline .github/workflows/milestone.yml:20]:
```suggestion
```

not needed?

File: .github/workflows/milestone.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+name: Check PR milestone
+
+on:
+  pull_request_target:
+    types: [milestoned, demilestoned, opened, reopened]
+    branches:
+      - main
```
Comment (AJPfleger) [inline .github/workflows/milestone.yml:7]: do we need to specify a branch, when we trigger on PR?

File: .github/workflows/milestone.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            if(context.payload.action === 'opened' || context.payload.action === 'reopened') {
+              if(milestone !== null) {
+                  core.notice(`Milestone is ${milestone.title}`);
+              } else {
+                const milestones = await github.rest.issues.listMilestones({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  state: "open"
+                });
+                for (const milestone of milestones.data) {
```
Comment (AJPfleger) [inline .github/workflows/milestone.yml:36]: I think `milestone` shadows the previous declaration on l.25. Shouldn't make a difference in this case, but maybe a bit dangerous

File: .github/workflows/milestone.yml (hunk 1 above)
Comment (paulgessinger) [reply .github/workflows/milestone.yml]: nope

File: .github/workflows/milestone.yml (hunk 2 above)
Comment (paulgessinger) [reply .github/workflows/milestone.yml:7]: I think this specifies the PR target branch in this context.

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [ca44b3fbaaee26fc7b9702da114d6f9fd698bf9e]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334243)||
|:green_circle: | [run_art_test: test_run4_acts_vertex_PU200](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334242)||
|:green_circle: | [run_art_test: test_run4_ttbar_PU200](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334241)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334240)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334239)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334238)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334237)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334236)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334235)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334234)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334233)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflowHgtd](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334232)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334231)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334230)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334229)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334228)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334227)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334226)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334225)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334224)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334223)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334222)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334221)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334220)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334219)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334218)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334217)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334216)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334215)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334214)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334213)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334212)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334211)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334210)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334209)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46334208)||

**Pull request merged**